google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -154,9 +156,21 @@ TracingMessageBatch::GetMessageSpans() const {
   return message_spans_;
```

Comment:
If `message_spans_` is guarded by `mu_`, then it seems like we need to lock `mu_` here.